### PR TITLE
OSSM-6154: skip tests for insignificant changes

### PR DESCRIPTION
**Pull request by FilipB**

### CODE REVIEW

File: ci-operator/jobs/maistra/istio-operator/maistra-istio-operator-maistra-3.0-presubmits.yaml [hunk 1]
```diff
@@ -1,7 +1,7 @@
 presubmits:
   maistra/istio-operator:
   - agent: kubernetes
-    always_run: true
+    always_run: false
```
Comment (jwendell) [inline ci-operator/jobs/maistra/istio-operator/maistra-istio-operator-maistra-3.0-presubmits.yaml:4]: Not sure about this. Do we need to change it? 
Wouldn't this mean we'd need to run all jobs manually?

I'd revert these changes, and keep only the `skip_if_only_changed` ones and experiment with it.

Comment (FilipB) [reply ci-operator/jobs/maistra/istio-operator/maistra-istio-operator-maistra-3.0-presubmits.yaml:4]: When using `skip_if_only_changed` the `always_run` must be false, otherwise it's failing with: "job pull-ci-maistra-istio-operator-maistra-3.0-unit is set to always run but also declares skip_if_only_changed targets, which are mutually exclusive"

Comment (jwendell) [reply ci-operator/jobs/maistra/istio-operator/maistra-istio-operator-maistra-3.0-presubmits.yaml:4]: I see. I hope prow is smart enough to run the jobs automatically when they don't fit in the `skip` rules :)

**Review by jwendell — APPROVED** [review]
